### PR TITLE
GCP build agent support

### PR DESCRIPTION
**Pull request by jroper**
Allows native images to be built on a GCP build agent, cutting build times in half.

Still to do:

* Make CI use this.
* Improve usage for running TCK against the native image locally.

### CODE REVIEW

**Comment by jroper** [comment]
@viktorklang you may wish to consider this if you want faster builds.

File: project/GraalVMPlugin.scala [hunk 1]
```diff
@@ -17,7 +16,7 @@ import com.typesafe.sbt.packager.graalvmnativeimage.GraalVMNativeImagePlugin.aut
  * Plugin to compile ahead-of-time native executables.
  *
  * @example Enable the plugin in the `build.sbt`
- * {{{
+ *          {{{
```
Comment (ihostage) [inline project/GraalVMPlugin.scala:19]: Looks like a random tap to the TAB key 😄

**Comment by viktorklang** [comment]
Thanks @jroper!

**Comment by jroper** [comment]
So, now that master is fixed, I've been able to run this against master (earlier my tests were all against a known working build from November last year), here's on my laptop:

```
[info] [cloudstate-proxy-core:30]    classlist:  15,666.71 ms,  2.85 GB
[info] [cloudstate-proxy-core:30]        (cap):     639.74 ms,  2.85 GB
[info] [cloudstate-proxy-core:30]        setup:   2,292.98 ms,  2.85 GB
[info] [cloudstate-proxy-core:30]   (typeflow): 130,203.14 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]    (objects): 109,030.44 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]   (features):   6,587.73 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]     analysis: 259,199.97 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]     (clinit):   6,619.70 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]     universe:  14,594.40 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]      (parse):  15,406.33 ms,  5.02 GB
[info] [cloudstate-proxy-core:30]     (inline):   9,215.56 ms,  5.46 GB
[info] [cloudstate-proxy-core:30]    (compile):  50,875.82 ms,  5.65 GB
[info] [cloudstate-proxy-core:30]      compile:  79,695.57 ms,  5.65 GB
[info] [cloudstate-proxy-core:30]        image:   8,282.54 ms,  5.70 GB
[info] [cloudstate-proxy-core:30]        write:   1,771.15 ms,  5.70 GB
[info] [cloudstate-proxy-core:30]      [total]: 382,037.27 ms,  5.70 GB
```

ie, 6 minutes 36 seconds, and here's on the GCP build agent:

```
[info] [cloudstate-proxy-core:49]    classlist:  16,406.49 ms,  3.20 GB
[info] [cloudstate-proxy-core:49]        (cap):     626.27 ms,  3.20 GB
[info] [cloudstate-proxy-core:49]        setup:   2,153.14 ms,  3.20 GB
[info] [cloudstate-proxy-core:49]   (typeflow):  49,907.00 ms,  7.94 GB
[info] [cloudstate-proxy-core:49]    (objects):  97,632.99 ms,  7.94 GB
[info] [cloudstate-proxy-core:49]   (features):   6,688.87 ms,  7.94 GB
[info] [cloudstate-proxy-core:49]     analysis: 168,552.56 ms,  7.94 GB
[info] [cloudstate-proxy-core:49]     (clinit):   7,759.49 ms,  7.97 GB
[info] [cloudstate-proxy-core:49]     universe:  11,869.85 ms,  7.97 GB
[info] [cloudstate-proxy-core:49]      (parse):   5,631.73 ms,  7.97 GB
[info] [cloudstate-proxy-core:49]     (inline):   4,914.13 ms,  7.67 GB
[info] [cloudstate-proxy-core:49]    (compile):  30,133.13 ms,  7.78 GB
[info] [cloudstate-proxy-core:49]      compile:  45,774.40 ms,  7.73 GB
[info] [cloudstate-proxy-core:49]        image:   7,759.34 ms,  7.78 GB
[info] [cloudstate-proxy-core:49]        write:   1,472.36 ms,  7.78 GB
[info] [cloudstate-proxy-core:49]      [total]: 254,513.50 ms,  7.78 GB
```

So 4 minutes 14 seconds. Not as nice a speed up as I measured earlier, looking at RAM usage, I suspect the reason might be that GraalVM RAM usage has improved, which means that performance on my laptop isn't as bad anymore. Still, it's cut more than a third off the time, and the good thing about using this in development is that you don't need to close Slack/browsers/IDEs etc while doing a native image compile (if I didn't close these before, the Linux OOM killer would usually kill the native image build).

**Comment by jroper** [comment]
Well it seems compilation times can be somewhat variable on the agent, I just got this:

```
[info] [cloudstate-proxy-core:49]      [total]: 211,667.70 ms,  7.31 GB
```

3 minutes 31 seconds. Much nicer.

**Comment by ralphlaude** [comment]
great job @jroper!

**Pull request merged**